.github/workflows/continous-integration.yml

```diff
@@ -99,6 +99,15 @@ jobs:
     - name: Lint Code 🎎
       run: make lint
 
+    - name: Checkout target branch to be able to diff
+      if: github.event_name == 'pull_request'
+      run: |
+        git fetch --depth=1 origin ${{ github.base_ref }}
+
+    - name: Check Docstring conventions 📃
+      if: github.event_name == 'pull_request'
+      run: make lint-docs branch=origin/${{ github.base_ref }}
```

Comment:
is there any way these checks can happen before the full `install poetry` stage? that way we wouldn't have to wait ten minutes before docstring errors are caught.

The only libs required for this are `flake8` and `flake8-docstrings` right?

Reply:
🤔 but why should this run before any other checks?